sdcm/results_analyze.py

```diff
@@ -663,3 +667,207 @@ def check_regression(self, test_id, is_gce=False):
+            # curr_test_stats = self._test_stats(row)
+            # if not curr_test_stats:
+            #     continue
+            if version not in group_by_version_sub_type:
+                group_by_version_sub_type[version] = dict()
+
+            if sub_type not in group_by_version_sub_type:
+                group_by_version_sub_type[version][sub_type] = dict(tests=SortedDict(), stats_best=dict(),
+                                                                    best_test_id=dict())
+                group_by_version_sub_type[version][sub_type]['stats_best'] = {k: 0 for k in self.PARAMS}
+                group_by_version_sub_type[version][sub_type]['best_test_id'] = {
+                    k: f"Commit: {version_info['commit_id']}, Date: {version_info['date']}" for k in self.PARAMS}
+            group_by_version_sub_type[version][sub_type]['tests'][version_info['date']] = curr_test_stats
+            old_best = group_by_version_sub_type[version][sub_type]['stats_best']
+            group_by_version_sub_type[version][sub_type]['stats_best'] = \
+                {k: self._get_best_value(k, curr_test_stats[k], old_best[k])
+                 for k in self.PARAMS if k in curr_test_stats and k in old_best}
+            # replace best test id if best value changed
+            for k in self.PARAMS:
+                if k in curr_test_stats and k in old_best and \
+                        group_by_version_sub_type[version][sub_type]['stats_best'][k] == curr_test_stats[k]:
+                    group_by_version_sub_type[version][sub_type]['best_test_id'][
+                        k] = f"Commit: {version_info['commit_id']}, Date: {version_info['date']}"
+
+        current_res_list = list()
+        versions_res_list = list()
+
+        test_version_info = self._test_version(doc)
+        test_version = test_version_info['version']
+        base_line = current_tests.get(subtest_baseline)
+        for sub_type in current_tests:
+            if not current_tests[sub_type] or sub_type == subtest_baseline:
+                self.log.info('No previous tests in the current version {} to compare'.format(test_version))
+                continue
+            cmp_res = self.cmp(current_tests[sub_type]['stats'],
+                               base_line['stats'],
+                               sub_type,
+                               current_tests[sub_type]['best_test_id'])
+            current_res_list.append(cmp_res)
+
+        if not current_res_list:
+            self.log.info('No test results to compare with')
+            return False
+        current_res_list = sorted(current_res_list, key=lambda x: x['version_dst'])
+        current_prometheus_stats = SortedDict()
+        for sub_type in current_tests:
+            current_prometheus_stats[sub_type] = {stat: current_tests[sub_type]["results"].get(
+                stat, {}) for stat in TestStatsMixin.PROMETHEUS_STATS}
+
+        for version in group_by_version_sub_type:
+            cmp_res = dict()
```

Comment:
Better to use SortedDict, then the order in Current_tests and and by version will be the same and not depend of order in filtered results from ES

Reply:
sort by what exactly ? we should sort them by time, and not alphabetically
anyhow dict are not order by insert order...

Reply:
in python 3 dict preserve the insert order
Sort by time is good idea, take it to cdc, thanks )